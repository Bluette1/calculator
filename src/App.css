#root {
  margin: 5% 5%;
}

@media (min-width: 992px) {
  #root {
    width: 700px;
    height: 800px;
    text-align: center;
  }
}
