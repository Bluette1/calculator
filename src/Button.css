button {
  flex-grow: 1;
  flex-basis: auto;
  font-size: 30px;
  border: 1px solid rgb(170, 167, 167);
}
