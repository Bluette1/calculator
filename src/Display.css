.display {
  height: 100px;
  background-color: rgb(128, 128, 128);
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 10px;
  color: rgb(255, 255, 255);
  font-weight: bold;
}

@media (max-width: 400px) {
  .display {
    display: flex;
    justify-content: space-evenly;
    height: 80px;
  }
}
