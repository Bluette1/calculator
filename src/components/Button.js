
import React from 'react';
const Button = (props) => {
    return (
      <div>
        <p>{props.value}</p>
      </div>
    )
};