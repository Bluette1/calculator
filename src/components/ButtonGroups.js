import React from 'react';
import PropTypes from 'prop-types';
import ButtonPanel from './ButtonPanel';

const ButtonGroups = props => {
  const { groups, onclick } = props;

  return (
    groups.map((btnPanel, idx) => (
      <React.Fragment key={Object.keys(btnPanel)[0]}>
        <ButtonPanel itemList={btnPanel[`itemList${idx}`]} handleClick={onclick} />
      </React.Fragment>
    ))
  );
};
ButtonGroups.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.object).isRequired,
};
ButtonGroups.propTypes = { onclick: PropTypes.func.isRequired };
export default ButtonGroups;
