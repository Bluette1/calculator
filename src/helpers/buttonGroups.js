const buttons = [
  { itemList: ['AC', '+/-', '%', 'รท'], key: 'group-1' },
  { itemList: ['7', '8', '9', 'x'], key: 'group-2' },
  { itemList: ['4', '5', '6', '-'], key: 'group-3' },
  { itemList: ['0', '.', '='], key: 'group-4' },
];
const buttonGroups = () => buttons;
export default buttonGroups;
