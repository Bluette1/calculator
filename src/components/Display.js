import React from 'react';
import PropTypes from 'prop-types';
import '../Display.css';

const Display = props => {
  const { result, error } = props;
  let display = result;
  if (error) {
    display = '\u26A0';
  }
  return <div data-testid="display" className="display">{display}</div>;
};
Display.defaultProps = {
  result: '0',
  error: false,
};

Display.propTypes = {
  result: PropTypes.string,
  error: PropTypes.bool,
};

export default Display;
