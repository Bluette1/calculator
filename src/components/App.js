import React from 'react';
import '../App.css';
import ButtonGroups from './ButtonGroups';
import buttonGroups from '../helpers/buttonGroups';
import Display from './Display';
import calculate from '../logic/calculate'

const App = () => (
  <>
    <Display />
    <ButtonGroups groups={buttonGroups()} />
  </>
);

export default App;
