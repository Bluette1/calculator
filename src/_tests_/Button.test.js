import React from 'react';
import ReactDOM from 'react-dom';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ButtonGroups from '../components/ButtonGroups';
import buttonGroups from '../helpers/buttonGroups';
import { shallow, mount } from "enzyme";
import App from '../components/App';
import Button from '../components/Button';
import Display from '../components/Display';

configure({ adapter: new Adapter() });

afterEach(cleanup);

it('renders the button component', () => {
  const btn = shallow(<Button value={'+'} />);
  expect(btn.text()).toBe('+');
});

describe("Button with the event handler", () => {
  const onClickSpy = jest.fn();
  it("displays button correctly",
    () => {
      const rendered = render(<Button value={'12'} onclick={onClickSpy} wide={false} color={'#f6f6f6'} />);
      expect(rendered).toMatchSnapshot();
      const renderedBtn = rendered.getByText("12");
      expect(renderedBtn).toMatchSnapshot();
      fireEvent.click(renderedBtn);
      expect(onClickSpy).toHaveBeenCalled();
      expect(renderedBtn.style.width).toBe('25%');
    }
  );
});

describe("Button with the wide prop set to true", () => {
  const onClickSpy = jest.fn();
  it("displays button correctly",
    () => {
      const rendered = render(<Button value={'12'} onclick={onClickSpy} wide={true} color={'#f6f6f6'} />);
      expect(rendered).toMatchSnapshot();
      const renderedBtn = rendered.getByText("12");
      expect(renderedBtn).toMatchSnapshot();
      fireEvent.click(renderedBtn);
      expect(onClickSpy).toHaveBeenCalled();
      expect(renderedBtn.style.width).toBe('50%');
    }
  );
});

describe("Button with the color prop set", () => {
  const onClickSpy = jest.fn();
  it("displays button correctly",
    () => {
      const rendered = render(<Button value={'12'} onclick={onClickSpy} wide={true} color={'#f6f6f6'} />);
      expect(rendered).toMatchSnapshot();
      const renderedBtn = rendered.getByText("12");
      fireEvent.click(renderedBtn);
      expect(onClickSpy).toHaveBeenCalled();
      expect(renderedBtn.style.backgroundColor).toBe('rgb(246, 246, 246)');
    }
  );
});

describe("Button without the color prop set - default color is used", () => {
  const onClickSpy = jest.fn();
  it("displays button correctly",
    () => {
      const rendered = render(<Button value={'12'} onclick={onClickSpy} wide={true} color={undefined} />);
      expect(rendered).toMatchSnapshot();
      const renderedBtn = rendered.getByText("12");
      fireEvent.click(renderedBtn);
      expect(onClickSpy).toHaveBeenCalled();
      expect(renderedBtn.style.backgroundColor).toBe('rgb(255, 123, 0)');
    }
  );
});