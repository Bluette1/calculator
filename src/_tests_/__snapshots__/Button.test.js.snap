// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with the color prop set displays button correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        style="width: 50%; background-color: rgb(246, 246, 246);"
        type="button"
      >
        12
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      style="width: 50%; background-color: rgb(246, 246, 246);"
      type="button"
    >
      12
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button with the event handler displays button correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        style="width: 25%; background-color: rgb(246, 246, 246);"
        type="button"
      >
        12
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      style="width: 25%; background-color: rgb(246, 246, 246);"
      type="button"
    >
      12
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button with the event handler displays button correctly 2`] = `
<button
  style="width: 25%; background-color: rgb(246, 246, 246);"
  type="button"
>
  12
</button>
`;

exports[`Button with the wide prop set to true displays button correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        style="width: 50%; background-color: rgb(246, 246, 246);"
        type="button"
      >
        12
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      style="width: 50%; background-color: rgb(246, 246, 246);"
      type="button"
    >
      12
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button with the wide prop set to true displays button correctly 2`] = `
<button
  style="width: 50%; background-color: rgb(246, 246, 246);"
  type="button"
>
  12
</button>
`;

exports[`Button without the color prop set - default color is used displays button correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        style="width: 50%; background-color: rgb(255, 123, 0);"
        type="button"
      >
        12
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      style="width: 50%; background-color: rgb(255, 123, 0);"
      type="button"
    >
      12
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
